core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,215 @@
+        //autopolling  start here
+        setStopPolling(false);
+        fluxDisposable = fluxHandle.subscribe(ps -> pollResponse = ps);
+    }
+
+    /**
+     * Constructor
+     *
+     * @param pollerOptions   poller options
+     * @param pollOperation   poller operation
+     * @param cancelOperation cancel operation
+     **/
+    public Poller(PollerOptions pollerOptions,
+                  Function<PollResponse<T>,
+                      PollResponse<T>> pollOperation,
+                  Consumer<Poller> cancelOperation) {
+        this(pollerOptions, pollOperation);
+        this.cancelOperation = cancelOperation;
+    }
+
+    private Flux<PollResponse<T>> createFlux() {
+        return sendPollRequestWithDelay()
+            .repeat(repeatPollTimes())
+            .takeUntil(pollResponse -> !isPollingStopped() && (pollResponse.status() == PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
+                || pollResponse.status() == PollResponse.OperationStatus.FAILED
+                || pollResponse.status() == PollResponse.OperationStatus.USER_CANCELLED));
+    }
+
+    /*Calculate how many time we should repeat poll*/
+    private int repeatPollTimes() {
+        // Poller options will ensure that TimeoutInMilliSeconds > 0
+        if (this.pollerOptions.getPollIntervalInMillis() > 0) {
+            return this.pollerOptions.getTimeoutInMilliSeconds() / this.pollerOptions.getPollIntervalInMillis();
+        } else {
+            // Poller Options validation will ensure that poll interval is > 0
+            // We should never reach here. This is extra safety guard.
+            return this.pollerOptions.getTimeoutInMilliSeconds();
+        }
+    }
+
+    /**
+     * This will cancel polling from Azure Service if supported by service
+     *
+     * @throws UnsupportedOperationException when cancel operation is not suported.
+     **/
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+        //We can not cancel an operation if it was never started
+        //or it is in its terminal state.
+        //Check cancelInitiated: to protect against multiple time call to cancel operation
+        if (cancelInitiated
+            || (pollResponse != null && pollResponse.status() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * This will poll and send PollResponse. If you had stopped polling erlier, we will enable polling again.
+     *
+     * @return Return poll response as Flux
+     **/
+    public Flux<PollResponse<T>> poll() {
+        setStopPolling(false);
+        if (fluxDisposable != null && !fluxDisposable.isDisposed()) {
+            fluxDisposable.dispose();
```

Comment:
@JonathanGiles  What Are your thoughts on, auto polling ? Because spec ask to start polling when we create Poller i.e. in constructor. When client also subscribe, we had to dispose the subscriber we created in constructor. Otherwise there will be two subscriber one created in constructor and second created by client. That is why I kind of not fan of Auto Polling
.

Reply:
Auto-polling is the current plan.